docs/api/browser-window.md

```diff
@@ -337,11 +337,12 @@ It creates a new `BrowserWindow` with native properties as set by the `options`.
       more details.
     * `contextIsolation` Boolean (optional) - Whether to run Electron APIs and
       the specified `preload` script in a separate JavaScript context. Defaults
-      to `false`. The context that the `preload` script runs in will still
-      have full access to the `document` and `window` globals but it will use
+      to `false`. The context that the `preload` script runs in will not
+      have full access to the `document` and `window` globals, will use
```

Comment:
This isn't accurate. They have access to the `document` and `window` objects, but they don't see the _same_ objects as the main content.

Think of it this way: in C++, there's a `Document` (and `Window`) object. To expose it to JavaScript, we create a wrapper object that translates JavaScript calls into C++ calls. When `contextIsolation` is on, there are _two_ wrapper objects: one for the preload script and one for the main content. Setting values on one `window` wrapper object doesn't make those values appear on the other wrapper object.

That isn't quite how it actually works, but it's a good enough analogy for this case.

Reply:
@bcpete I feel like you could copy-edit @nornagon's explanation and just add _that_ to the docs 😉

Reply:
I understand, I feel like the wording is misleading. When I read "full access to the document and window globals" it makes me think I can alter the globals for the main content by mutating the window object within the preload script. Like you pointed out though this is not the case as they have different contexts.

Which was the whole point of adding contextIsolation I'd assume, it's kind of in the name, I just think for anyone digging through the docs trying to find a solution to fit their use case this section is a little bit misleading.

Any suggestions on how to improve my changes? Maybe I should change the section to better describe what context isolation is doing, and point towards solutions for mutating the main contents globals?

Reply:
Sounds good, thanks for the feedback!

Reply:
Pushed some changes, let me know if you'd like anything added.